lib/workers/repository/process/lookup/index.ts

```diff
@@ -541,7 +541,15 @@ export async function lookupUpdates(
     // Add digests if necessary
     if (supportsDigests(config.datasource)) {
       if (config.currentDigest) {
-        if (!config.digestOneAndOnly || !res.updates.length) {
+        let alwaysUpdateDigest = true;
+        if (config.digestOneAndOnly === true) {
+          alwaysUpdateDigest = false;
+        } else if (config.datasource.startsWith('custom.')) {
+          // Custom datasources should not run a digest update
+          // on the current version if it already runs a version update.
```

Comment:
If it "already ran a version update" then wouldn't res.updates.length be non-zero length anyway?

Reply:
then this isn't the right solution, because it should be possible to do digest only updates with regex manager. I do that heavily on jenkinsfiles for docker images

Reply:
My solution here still allows for digest-only updates. My example repo covers scenarios for digest-only updates. The unit tests pass as well, which contain digest-only updates.

The problem was that renovate will do a digest-only additionally to a version + digest update for the same dependency. Under the circumstances above, this will result in an incorrect warning that renovate cannot look up a digest.